orm/serial_bucket.go

```diff
@@ -0,0 +1,358 @@
+package orm
+
+import (
+	"reflect"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+)
+
+// SerialModel is implemented by any entity that can be stored using SerialModelBucket.
+// GetID/SetID are used to store and access the Key.
+// The ID is always set to nil before serializing and storing the Value.
+type SerialModel interface {
+	weave.Persistent
+	Validate() error
+	GetID() []byte
+	SetID([]byte) error
```

Comment:
Interesting approach.

This does assume that the primary key is always prepared and I think it may push some more work to the users of the API.

Note that `GetID` is auto-generated from protobuf, so this was only one method in either case.

I do like the rename to `PrimaryKey` over `ID` as well.